portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,25 +28,42 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data using Datastore */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   private final ArrayList<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(comments);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity commentEntity : results.asIterable()) {
+      String comment = (String) commentEntity.getProperty("comment");
+      String name = (String) commentEntity.getProperty("name");
```

Comment:
It doesn't look like these are being used anywhere. Please update or add a TODO comment.

Reply:
Oops - I forgot to add a line below that adds these variables into the comments list. I'll fix that.